src/stores/media/index.ts

```diff
@@ -385,6 +387,11 @@ export const useMediaStore = defineStore('media', {
                 error.response?.status ?? 'unknown'
               }`
       } else {
+        /*Capture the error and all of its details using $sentry */
+        Sentry.captureEvent({
```

Comment:
Inside of Pinia stores you can access the [pre-configured Sentry instance via `this.$nuxt.$sentry`](https://sentry.nuxtjs.org/). In this case I think it'd make sense to use `captureException` explicitly when an error is caught. For the case where we set the message to "Oops! Something went wrong", if we could capture the event with something more specific that would be great. Maybe `captureMessage` is better in that case. I'm not totally sure of the difference. Do you happen to know?

@obulat and @dhruvkb do y'all know when that "Oops! Something went wrong" clause happens and what it's meant to handle?

BTW, to test this, I threw an arbitrary error in fetchSingleMediaType before `setMedia` is called and then a breakpoint in my browser debugger on this line.

Reply:
> Inside of Pinia stores you can access the [pre-configured Sentry instance via `this.$nuxt.$sentry`](https://sentry.nuxtjs.org/).
I tried accesing it this way before but it keeps failing when i run pnpm test with the error `"Cannot read properties of undefined (reading 'captureException')"`

Reply:
Ah! You've got to mock out context variables in tests. You can add $nuxt and the $sentry mock to this Pinia test utility we use: https://github.com/WordPress/openverse-frontend/blob/b863e3e4b88eb667a87a2c620abc688ab29838eb/test/unit/test-utils/pinia.js#L4-L9

Reply:
> if we could capture the event with something more specific that would be great. Maybe captureMessage is better in that case. I'm not totally sure of the difference. 

[According to sentry docs,](https://docs.sentry.io/platforms/javascript/usage/) we can pass a non-error object to captureException tho it will lack the context. `You can pass an Error object to captureException() to get it captured as event. It's also possible to pass non-Error objects and strings, but be aware that the resulting events in Sentry may be missing a stacktrace.`

Capture event triggers a manual event, that means we recieve all the breadcumbs that were collected until that moment. We can also include a dump of the application if we are sure we won't surpass the [event payload max size](https://develop.sentry.dev/sdk/event-payloads/)

Reply:
For now I think better not to dump the application. If we need it to debug a particular error that sounds like a great feature though, thanks for sharing it.

Do whatever you think is best _vis a vis_ which Sentry function to use :slightly_smiling_face:

Reply:
> @obulat and @dhruvkb do y'all know when that "Oops! Something went wrong" clause happens and what it's meant to handle?

No, I don't know what it was meant to handle. It was there even before Nuxt :) I really wonder what we catch with this PR!

Reply:
Then i believe capture event is the best option, [as we can enrich the event data as we see fit ](https://docs.sentry.io/product/sentry-basics/integrate-backend/capturing-errors/)